src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/HearingDocumentService.java

```diff
@@ -91,9 +95,15 @@ CaseDetails addCourtFields(CaseDetails caseDetails) {
     }
 
     public void sendFormCAndGForBulkPrint(CaseDetails caseDetails, String authorisationToken) {
-        List<BulkPrintDocument> caseDocuments = getHearingCaseDocuments(caseDetails.getData());
-        bulkPrintService.printApplicantDocuments(caseDetails, authorisationToken, caseDocuments);
-        bulkPrintService.printRespondentDocuments(caseDetails, authorisationToken, caseDocuments);
+        String caseId = caseDetails.getId() == null ? "noId" : caseDetails.getId().toString();
+        List<BulkPrintDocument> caseDocuments = getHearingCaseDocuments(caseDetails.getData(), caseId);
+
+        if (!notificationService.shouldEmailContestedAppSolicitor(caseDetails.getData())) {
+            bulkPrintService.printApplicantDocuments(caseDetails, authorisationToken, caseDocuments);
+        }
+        if (!notificationService.shouldEmailRespondentSolicitor(caseDetails.getData())) {
```

Comment:
that method name deserves a refactor to the java standards, when returning boolean the method should start with "is"